embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/injection/ModuleInitBootstrapper.kt

```diff
@@ -478,7 +478,7 @@ internal class ModuleInitBootstrapper(
     }
 
     fun stopServices() {
-        if (isInitialized()) {
```

Comment:
i'm guessing we check this since you can't have `asyncInitTask` be `null` when it is used with `synchronized`, but this i think it would be more readable if that was check was done directly. or add a comment explaining that that is why this is checked here and then again below.

that said, is this 100% safe? in theory, couldn't `asyncInitTask` become `null` after the initial check? should we just get rid of the initial check on L481 and use some other object in the `synchronized` statement?

Reply:
I added to prevent unnecessary locking under high contention situations. In practice, it shouldn't matter in this specific case, but it's an idiom I follow whenever I lock.

The first check isn't necessarily threadsafe, but it's not supposed to be - it's just an optimization to do an early return and not have to lock. The second check is threadsafe, and that's all that matters for that.